rewrite pyopenms-wheels to use matrix builds

poshul opened the pull request:
## Description

<!-- Please include a summary of the change and which issue is fixed here. -->

## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.seqan.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>
  
- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds
  
</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

timosachsenberg committed:
resolve merge conflicts

timosachsenberg committed:
more path changes

timosachsenberg committed:
last paths

timosachsenberg committed:
fix some tests

timosachsenberg committed:
more tests

timosachsenberg committed:
more tests

timosachsenberg committed:
nop

timosachsenberg committed:
fix more annotation tests

timosachsenberg committed:
added some debug code that shows an issue with the check for amidation in deisotoper

timosachsenberg committed:
update DecoyGenerator test

timosachsenberg committed:
update nuXL tests

timosachsenberg committed:
factor special functionality of NuXLDeisotoper out

timosachsenberg committed:
fix SSE test

timosachsenberg committed:
remove feature augmentation

timosachsenberg committed:
remove some unused functionality

timosachsenberg committed:
Apply suggestions from code review

timosachsenberg committed:
add missing include

timosachsenberg committed:
Merge remote-tracking branch 'origin/feature/NuXL' into feature/NuXL

timosachsenberg committed:
missing test files

timosachsenberg committed:
partial loss score still different

timosachsenberg committed:
partial loss score still different

timosachsenberg committed:
Merge remote-tracking branch 'origin/feature/NuXL' into feature/NuXL

timosachsenberg committed:
removed some compile warnings

timosachsenberg committed:
more consistent ion charge format

timosachsenberg committed:
remove annotation of internal ions (to many duplicated m/z)

timosachsenberg committed:
increase determinism

timosachsenberg committed:
stupid windows

timosachsenberg committed:
Merge remote-tracking branch 'origin/develop' into feature/NuXL

timosachsenberg committed:
add tests and documentation for deterministic fragment adduct generation

timosachsenberg committed:
Merge branch 'develop' into feature/NuXL

cbielow committed:
Merge branch 'develop' of https://github.com/OpenMS/OpenMS into develop

cbielow committed:
Merge branch 'develop' of https://github.com/OpenMS/OpenMS into develop

timosachsenberg committed:
Add support for custom RNA/DNA presets in json file (#7856)

* add json

* minor error check

* move file to share

* attempt to move all presets to json

* remove built in presets

* remove debug output

* add advanced option to OpenNuXL to specify custom presets file

* Update README.md with preset usage instructions

timosachsenberg committed:
some additional documentation

timosachsenberg committed:
remove some unused code

subCode321 committed:
changed minimum header size

poshul committed:
rewrite pyopenms-wheels to use matrix builds

poshul committed:
Fix deb dependencies (#7943)

qt6-base-dev not required for OpenMS usage

poshul committed:
fix os names for conda install

poshul committed:
specify container

poshul committed:
fix os

poshul committed:
pass correct var

poshul committed:
remove *

poshul committed:
Specifiy bash on windows

poshul committed:
fix path for auditwheel

poshul committed:
make pyopenms_whls relative to workspace

poshul committed:
try a different path fix

poshul committed:
try unit tests without deactivating conda

poshul committed:
Try cibuildwheel

poshul committed:
tell cibuildwheel where the root is

poshul committed:
add pip to manylinux install

poshul committed:
populate pyproject.toml from setup.py

poshul committed:
add docker to manylinux skip python 3.8

jpfeuffer commented on pyproject.toml:
```diff
@@ -1,3 +1,52 @@
+requires = [
+    "setuptools",
+    "wheel",
+    "scikit-build",
```
I dont think we need scikit build. Was just copied.

poshul committed:
fix stuff

poshul committed:
remove package install for ubuntu

poshul committed:
don't use py 3.8 or older

poshul committed:
set package version from  environmental variable using setuptools_scm

poshul committed:
increase cibuildwheel verbosity

poshul committed:
underscores

poshul committed:
toml formatting

poshul committed:
try copy cibuildwheel.toml on linux

poshul committed:
move contents of cibuildwheels.toml file to pyproject.toml

poshul committed:
syntax

poshul committed:
-

poshul committed:
fix ctest command

poshul committed:
correct skip

poshul commented:
Switching to CIBuildTools looks more complicated than expected. We already roll the wheel building process into our ctest call, but as far as I can tell there is no way to disable CIBuildTools calling either `pip` or `build` to generate the wheels. We can call our ctest build command in the `before-build` section of the definition, which succeeds, but then chokes on calling `pip wheel` after.

As far as I see it the correct way to fix this would be to decouple the actual wheel building from the ctest command. @jpfeuffer, does that seem reasonable?

jpfeuffer commented:
I think the cleanest solution is to trigger CMake with setuptools. See my efforts in better_pyopenms_build. 

You could quickly try if you can add `--no-binary pyopenms` to skip building binaries after the prebuild. Like we do in our Cmake.
https://github.com/OpenMS/OpenMS/blob/develop/src/pyOpenMS/CMakeLists.txt#L510

cbielow committed:
make GaussFitter a lot faster

jpfeuffer commented:
I am not sure it chokes because the wheel was already built. It probably chokes because it tries to build the wheel again but with the setup.py instructions which are completely broken and outdated.

cbielow committed:
Merge branch 'develop' of https://github.com/OpenMS/OpenMS into feat_ppfast

jpfeuffer commented:
Depending on if cibuildwheel reuses the build folder you might want to build (lib)Openms before, install it globally and then run the CMake part for pyopenms triggered via setup.py.

subCode321 committed:
Fixed logic of parse

subCode321 committed:
Merge branch 'develop' into fixcrash

cbielow committed:
Speed improvement to GaussTraceFitter (#7950)

* fix error in concatenation

* make GaussFitter a lot faster

* mod CHANGELOG

singjc committed:
DOC: Minor update to OSW TOPP doc (#7956)

* update: OSW doc usage of use_ms1_traces to enable_ms1

* update: example spyo chrom mzML param use_ms1_traces to enable_ms1

subCode321 committed:
drop changes to check for valid headers

subCode321 committed:
Merge branch 'fixcrash' of github.com-subcode:subCode321/OpenMS into fixcrash

cbielow committed:
Merge branch 'develop' into fixcrash

cbielow committed:
fixes #7923 (Targetedfileconverter crash)

BugFix : Fixed Targetedfileconverter crash

timosachsenberg committed:
improve compatibility with mzTabM standard (#7958)

* minor fixes for MzTab

* choose more sane defaults

* fix test

ChandraDevs1 committed:
FastaFile cannot start with an empty line (#7960)

* Enhanced test coverage in

* added changelog for new updates

* fixes the issue for #7839

subCode321 committed:
changed implementation

subCode321 committed:
Revert "changed implementation"

This reverts commit 94e59bdd63eff691a7f04bdbcd5774c35895648f.

subCode321 committed:
add const and const iterator

timosachsenberg committed:
Update PercolatorInfile.cpp to handle a corner case in sage (#7977)

* Update PercolatorInfile.cpp

* Apply suggestions from code review

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

---------

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

subCode321 committed:
add logic to find minima

subCode321 committed:
add min finder utility

subCode321 committed:
Merge branch 'develop' into alignwork

subCode321 committed:
use getScoreComparator static const function

subCode321 committed:
Merge branch 'alignwork' of github.com-subcode:subCode321/OpenMS into alignwork

subCode321 committed:
document higher score better

subCode321 committed:
test running after change

subCode321 committed:
rename param

subCode321 committed:
fix param docs

cbielow committed:
Merge pull request #7973 from subCode321/alignwork

Made align function work on const data

t0mdavid-m committed:
Enable OpenMP for pyOpenMS on MacOS (#7980)

* add test run

* push on dev

* create contrib folder

* Delete .github/workflows/test_contrib.yml

* disable openmp for arm wheels

* set rpath

* try fix from abibuilder

* switch linking

* fix typo

* add print for libraries

* fix openmp build flags

* fix rpath in workflow

* dont remove source

* install wheels

* run unit test separately

* Revert "install wheels"

This reverts commit 7c159d2bc4b3566eb3e8ccfd433bf58facf410ab.

* Revert "run unit test separately"

This reverts commit 5b854cb4a537379d3e3e3d5d06d3f9e9e8cef398.

* Reapply "install wheels"

This reverts commit 0f1ff8dbe0f0c0df506d8d955b7692b86641a05f.

* Revert "dont remove source"

This reverts commit 227b3a96ab10f1c9baecc4149b891c1277de4d19.

* is brew libomp the issue?

* update rpath

* rpath to lib

* update cmake args

* remove lib

* move test files to tests

* move test files

* dont uninstall brew

* cleanup

* fix typo

* clean up libs

* cleanup

timosachsenberg committed:
refactor: move basename member out of PeptideIdentification into MetaValues (#7981)

* move basename member out of PeptideIdentification into MetaValues

* fix tests and simplify them

* remove base name (not needed)

* strange character fix

timosachsenberg committed:
doc: PeptideHit and PeptideIdentification (#7983)

* improve doc, add type alias

* Add deprecation notice for PeptideHit class

* Deprecate PeptideIdentification in favor of SpectrumIdentification

poshul committed:
Update THIRDPARTY and lay groundwork for linux ARM support (#7811)

* Add ubuntu arm to matrix

* Update thirdparty with new percolator and some arm64 stuff

* Update thirdparty, update Percolator version

* Update percolator test for 3.07 (NB the actual Percolator 3.07 release still uses 3.06 in their executable)

* Add logic to copy the correct THIRDPARTY executables for the current arch

* Update Comet versions in all OSs update comet test files for the new version.

* add status of copy

* cp issues for macZZ

* Update openms_ci_matrix_full.yml

Comment out the actual ARM run for now so we can merge the updated THIRDPARTY

* third

timosachsenberg committed:
type alias for MSexperiment (MSRun) (#7988)

* reserve a type alias

* small additions to architecture file

* Documents manual Python extension capabilities

Adds a detailed section on using manual PYX files to extend Python bindings, covering advanced functionality such as handling complex C++ features, Python-specific enhancements, performance optimizations, and integration with Python libraries. Clarifies the role of PYX files in enabling features beyond automatic wrapping.

* Update ARCHITECTURE.MD

timosachsenberg committed:
refactor: move significance_threshold and pepXML analysis results member in PeptideIdentifications to a meta value as it is often unused (#7985)

* improve doc, add type alias

* move pepxmlanalysis results into meta values to simplify datastructure

* move significant threshold to metavalue

* some fixes

* Merge remote-tracking branch 'origin' into openms-pr-5547

* more test updates

* more fixes

* don't write it twice

* tests works

* fix tests

* more test fixes

* nop

cbielow committed:
Merge branch 'develop' of https://github.com/OpenMS/OpenMS into develop

cbielow committed:
exclude dll from pre 3.4 installer causing cxxframehandler errors when running TOPP executables on Win10

cbielow committed:
Installer dll fix (#7995)

* fix error in concatenation

* make GaussFitter a lot faster

* exclude dll from pre 3.4 installer causing cxxframehandler errors when running TOPP executables on Win10

cbielow committed:
use VS2022 (according to https://github.com/jurplel/install-qt-action?tab=readme-ov-file#arch) to make Qt6 build using the same compiler

cbielow committed:
add more docs to Windows Compiler versions

cbielow committed:
Merge branch 'develop' into installer_dll_fix

cbielow committed:
Merge branches 'develop' and 'develop' of https://github.com/OpenMS/OpenMS into develop

cbielow committed:
fix Windows dll exclude regex (CMake regexes are case sensitive and the extracted system32 paths have weird cases)

cbielow committed:
Merge pull request #8001 from cbielow/fix_dll_install

Fix Win Installer DLL Exclusion

pjones committed:
[TEST] Make the BilinearInterpolation test deterministic (#8000)

The BilinearInterpolation test did not properly initialize all of the
matrix elements before comparing for equality.  Since NaN does not
compare with itself, the test would randomly fail when one or more
elements were initialized to NaN.

I also removed unused headers as reported by clangd.

poshul committed:
Update openms_ci_matrix_full.yml (#8003)

* Update openms_ci_matrix_full.yml

* Update openms_ci_matrix_full.yml

* Update openms_ci_matrix_full.yml

* Update openms_ci_matrix_full.yml

Apply suggested formatting

jpfeuffer committed:
Merge branch 'develop' into installer_dll_fix

jpfeuffer committed:
Merge pull request #7996 from cbielow/installer_dll_fix

Window Installer Qt6 fix

ketan1406 committed:
Update MacOS bg images & Windows splash image with new OpenMS Logo; Solves part of Issue #7891 (#7922)

* Update MacOS background images and Windows splash image with new OpenMS logo

* Update: square logos to blackFont

---------

Co-authored-by: Samuel Wein <sam@samwein.com>

poshul committed:
Update THIRDPARTY with new comet (#8012)

Co-authored-by: Timo Sachsenberg <timo.sachsenberg@uni-tuebingen.de>

cbielow committed:
fix #8014: opening a single MS2 spec in 3D view should not be allowed (3D view does not support MS2 spectra)

cbielow committed:
- avoid redrawing when filters are initially set while setting up a new layer
 - avoid redrawing when filters actually do not change

timosachsenberg committed:
Range exception (#8017)

* throw exception in range base

* fix test

* use new macros

* only use isEmpty()

* isEmpty

* Removes redundant RangeManager tests

Eliminates tests for default min/max range values in RangeManager, as they are no longer necessary or relevant.

* fix test

* forgot mz and rt

* doc

* in memory mzML updates ranged on load

* fix class tests

* fix topp tests

* fix last tests

* Adds assertions to verify the necessity of range updates and logs warnings when updates are redundant.

* removed duplicate includes

ZenMachina16 committed:
Handled Single-Peak Mass Traces (#8011)

* Handled Single-Peak Mass traces

* bit detailed

* Update src/openms/source/KERNEL/MassTrace.cpp

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

---------

Co-authored-by: Timo Sachsenberg <timo.sachsenberg@uni-tuebingen.de>
Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

poshul committed:
use pattern for download-artifact (#8009)

* use pattern for download-artifact

implement https://github.com/OpenMS/OpenMS/issues/8004

* Update .github/workflows/openms_ci_matrix_full.yml

Co-authored-by: Julianus Pfeuffer <8102638+jpfeuffer@users.noreply.github.com>

---------

Co-authored-by: Julianus Pfeuffer <8102638+jpfeuffer@users.noreply.github.com>

poshul committed:
Merge branch 'pyopenms_better_build' into cleanup_pyopenms_wheels

adityamuzumdar committed:
Add option to PeakFileOption to ignore loading (#7901) (#7902)

* Add option to PeakFileOption to ignore loading (#7901)

* refactor

* make changes to mzmlhandler and update test

* remove chromatogram enum

* refactor

* update authors and changelog

* added comments

* negate condition

---------

Co-authored-by: Muzumdar <aditya.muzumdar@hilti.com>
Co-authored-by: Timo Sachsenberg <timo.sachsenberg@uni-tuebingen.de>

cbielow committed:
Merge branch 'develop' into fix_TV3d

poshul committed:
Cleanup

poshul committed:
dont try to build for 3.6

poshul committed:
fix syntax

poshul committed:
fix license to pre pep 639

poshul committed:
fix stuff

poshul committed:
try more formatting shit

poshul committed:
fixxxxxxxxx

poshul committed:
try repeating the "skip" for tool.cibuild.linux

poshul committed:
move skip to env var, linux seems to ignore it otherwise.

poshul committed:
reduce verbosity to sane amount

poshul committed:
specify cibuildwheel toml loc

poshul committed:
correct path for toml

poshul committed:
config flag change

poshul committed:
don't build win32

poshul committed:
try making before-build single line

poshul committed:
Pass source_directory to docker containers

poshul committed:
set up vsshell for windows

poshul committed:
try passing image path for SOURCE_DIR

timosachsenberg committed:
added update ranges to other file formats (#8020)

pjones committed:
Merge branch 'develop' into fix_TV3d

pjones committed:
Merge pull request #8019 from cbielow/fix_TV3d

TOPPView: open theoretical spectra immediately

poshul committed:
export correct path to contrib in linux, explicitly tell ctest which python to use

poshul committed:
fix path and varname

poshul committed:
use our linux image, correct where python exec is set

poshul committed:
fix quote escaping

poshul committed:
try exporting python exec as separate call

poshul committed:
linuxvar passing stuff

poshul committed:
fix

poshul committed:
only use the venv to find python

poshul committed:
define CIBW_ENVIRONMENT_PASS_LINUX after updating SOURCE_DIRECTORY

jpfeuffer commented on src/pyOpenMS/setup.py:
```diff
@@ -1,238 +1,324 @@
+                             "be 'release', 'debug' or 'relwithdebinfo'")
+
+        # The directory containing this setup.py
+        source = os.path.dirname(os.path.abspath(__file__)) + "/../../"
```
I think ideally CMake should be run directly from the pyopenms directory. It will then try to find OpenMS via find_package (which you need to somehow prebuild).
Since we install Cmake configs we also export our compile flags etc.

Maybe try it after it successfully builds. There might be a bit less configuration settings to do in the setup.py when doing it that way. Plus saving time. Plus, we can more easily separate pyopenms.

poshul committed:
Make sure classes that use rangeManager aren't empty during pyopenms … (#8022)

* Make sure classes that use rangeManager aren't empty during pyopenms tests

* reorder python tests

pjones committed:
[DOC] Be specific about Boost flags and improve formatting (#8023)

Boost installed from brew requires dynamic linking.  This is mentioned
in the install documentation but without saying which flag to pass to
CMake.  This commit resolves that.

It also fixes the formatting for a code block.  The indentation under
the <li> element needs to remain consistent otherwise the \code
markers are ignored and the entire section is treated as code.

Krishnadubey1008 committed:
simplifies the Linux build instructions specifically for Ubuntu (#7897)

* Simplify Linux build instructions for Ubuntu (#7294)

* Simplify Ubuntu Linux build instructions in Doxygen docs (#7294)

* removed .md file as requested by the maintainer

* Improve structure and clarify choice between make and ninja (#7294)

* Update doc/doxygen/install/install-linux.doxygen

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

---------

Co-authored-by: krishna dubey <krishnadubey1008@gmail.com>
Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>
Co-authored-by: Samuel Wein <sam@samwein.com>

Son-0f-Anton committed:
fix documentation inconsistency (#7971)

Co-authored-by: Samuel Wein <sam@samwein.com>

poshul committed:
Update install-linux.doxygen (#8027)

pjones committed:
Update for Release 3.4 (#8028)

Generate a new `CHANGELOG_PARAMS` file for the release.

Co-authored-by: Tom David Müller <57191390+t0mdavid-m@users.noreply.github.com>

github-actions[bot] committed:
Updated OpenMS package version number (#8033)

Co-authored-by: t0mdavid-m <t0mdavid-m@users.noreply.github.com>

poshul committed:
Merge 3.4.0 into develop (#8037)

* Update openms_ci_matrix_full.yml

Fix destination paths for KNIME artifact downloads.

* Update package_mac_productbuild.cmake

Change logo scaling for installer to "tofit"

* Update FLASHDeconvWizardBase.cpp (#8034)

updated link to flashdeconv web site

* Replace neckarfront with logo

* Update package_mac_productbuild.cmake

Disable mac installer logo

* Fix debian installer (#8035)

* add libyaml dependency

* proper fix

* Update openms_ci_matrix_full.yml

fix download-artifact issue

---------

Co-authored-by: Tjeerd Dijkstra <tjeerd.dijkstra@uni-tuebingen.de>
Co-authored-by: Tom David Müller <57191390+t0mdavid-m@users.noreply.github.com>

cbielow committed:
TOPPAS: tutorial update (#8025)

* - new TOPPAS tutorial section on transferring parameters and some cleanup
 - TOPPAS input file nodes report file types according to how OpenMS sees the files now

* fix doxygen warnings

* update CHANGELOG (also for older PRs)

timosachsenberg committed:
refactor: reduce the use of assignRanks (#7991)

* reduce the use of assignRanks

* Apply suggestions from code review

* removes sortByRank

* fix mzid test

* update  changelog

pjones committed:
Resolve undefined behavior w.r.t. basic_string<unsigned_char> (#8024)

`std::char_traits<unsigned char>` has been undefined behavior since at
least C++11 and was recently removed in LLVM/clang.  Therefore the
latest version of XCode no longer allows:

    std::basic_string<unsigned char>

This means that OpenMS will no longer compile on macOS after recently
released system updates are installed.

I "fixed" the issue by replacing `std::basic_string` with
`std::string` and a cast.  Since SQLite is returning a UTF-8 character
string, there's a chance that string comparisons will not work the way
we expect if wide characters are returned.

I still need to verify that the `OSWFile` test is actually checking
the section of code changed in this commit.

pjones committed:
[DOC/CI] Make installation documentation and CI use the same commands

Prevent the documentation from becoming stale by using the same
dependency installation commands as CI.

timosachsenberg committed:
refactor: peptide id - move rank to metavalue (#7997)

* reduce the use of assignRanks

* Apply suggestions from code review

* removes sortByRank

* fix mzid test

* move rank to meta value

* update tests

* test files

* move constant to Constants.h

* fix pyopenms

* update changelog

* update test

* update topp tests

* update topp tests

* one more test

* switched to new logic

* forgot one file

* sage

* nop

* nop

* last

* fix for sage

* removed default rank

cbielow committed:
Show file load/store progress in all TOPP tools (#8041)

* correctly forward TOPP's log_type to FileHandler calls

* augment CHANGELOG

timosachsenberg committed:
Add doc (#8044)

* add missing kernel doc

* more doc

cbielow committed:
avoid "unknown pragma" warnings on MSVC (#8045)

timosachsenberg committed:
refactor: FFIDmoves many of the external id related functionality in separate class. (#8016)

* start separating external id stuff

* move svm code into externid class

* moved more

* simplify a bit more

* nop

* removed unused functions

* Adds private method suffix for consistency

Renames private methods in FFIDAlgoExternalIDHandler with a trailing underscore to follow naming conventions and improve code readability. No functional changes introduced.

* Namespaces FFIDAlgoExternalIDHandler for clarity

Encapsulates FFIDAlgoExternalIDHandler within the Internal namespace to improve code organization and clarify its intended scope of use. Updates relevant references and adds the required namespace import.

Improves code modularity and maintainability.

* Fix namespace closure for FFIDAlgoExternalIDHandler

* Update src/openms/source/FEATUREFINDER/FFIDAlgoExternalIDHandler.cpp

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

---------

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

poshul committed:
Cleanup tags (#8036)

* Update openms_ci_matrix_full.yml

Fix destination paths for KNIME artifact downloads.

* Update package_mac_productbuild.cmake

Change logo scaling for installer to "tofit"

* Update FLASHDeconvWizardBase.cpp (#8034)

updated link to flashdeconv web site

* Replace neckarfront with logo

* Update package_mac_productbuild.cmake

Disable mac installer logo

* Fix debian installer (#8035)

* add libyaml dependency

* proper fix

* Update openms_ci_matrix_full.yml

fix download-artifact issue

* rework tag creation to be less cryptic, add action to publish draft release

* fix knime makr latest target

* Update openms_ci_matrix_full.yml

---------

Co-authored-by: Tjeerd Dijkstra <tjeerd.dijkstra@uni-tuebingen.de>
Co-authored-by: Tom David Müller <57191390+t0mdavid-m@users.noreply.github.com>
Co-authored-by: Timo Sachsenberg <timo.sachsenberg@uni-tuebingen.de>

cbielow committed:
TOPPView: fixes a few error messages boxes (im IM frames, and 2D projections) (#8047)

* avoid "unknown pragma" warnings on MSVC

* fix TV error message box when right-clicking on a 2D mobilogram (which has no RT information)

* avoid error in TV when activating 2D projections (for any type of data)

* fix: allow projections for IM frames from MS2 data (would otherwise be empty)

timosachsenberg committed:
refactor: separate ranges (#8021)

* added update ranges to other file formats

* explorative attempt

Introduces dual-range management for spectra and chromatograms

Replaces the single RangeManager with separate SpectrumRangeManager
and ChromatogramRangeManager, enabling independent handling of ranges
for MS spectra and chromatograms. Maintains backward compatibility
using a combined range manager. Adds extensive tests to validate
the new dual-range system and its integration with existing functionality.

This change aims to improve modularity and clarity in range handling,
while preserving legacy behavior for seamless integration.

* some progress

* fix

* more fixes

* Adds detailed range reporting for MSExperiment data

Introduces specialized templates for human-readable and machine-readable range reporting in MSExperiment, covering retention time, mass-to-charge, ion mobility, and intensity across combined, spectrum-specific, and chromatogram ranges. Enhances the clarity and granularity of data outputs.

* more fixes

* more fixes

* some fixes

* compiles again

* fix tests

* add file

* add cos

* add python bindings

* make private

* simplify interface

* simplify a bit more

* update test

* address reviewer comments

* remove non const access

* don't treat chromatograms as metadata

* exclude precursors from range calculation

pjones committed:
Pjones/backports (#8051)

* [CI] Extract the version update code into its own script

* [Docker] Use the correct THIRDPARTY directory name

---------

Co-authored-by: Samuel Wein <sam@samwein.com>

jcharkow committed:
fix: r^2 is zero (#8052)

r^2 is zero because it is not computed. Causes downstream OpenSwath to
fail. Ensure goodness of fit is set to true so r^2 is computed

SGSSGene committed:
Feat/option to toggle tdl (#8055)

* patch: rename ENABLE_CWL to ENABLE_CWL_GENERATION

* feat: add ENABLE_TDL option to enable/disable TDL dependency

timosachsenberg committed:
refactor: Store IDs outside of spectra.  (#7974)

* Add AnnotatedMSRawData and integrate it instead of MSExperiment where identification is needed

* Various refactorings in order to integrate the new class "AnnotatedMSRawData"

Build target "OpenMS" is successfully compiling now.

* some fixes to tests

* remove remnantsof protein identification at spectra

* Apply suggestions from code review

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

* minor

* fix test

* class and tool tests pass

* more fixes

* rename

* removed commented out code

* Update src/openms_gui/source/VISUAL/PlotCanvas.cpp

* Update src/openms/source/ANALYSIS/ID/IDMapper.cpp

* nop

* simplified interface a bit more

* moved file out of kernel

* remove mzML from IDSplitter

* forgot one file

* removed unused mapping class

* compiles again

* some TV fixes

* More identification view fixes

* peak annotations work again

* more safety checks

* more imp to cpp

* mend

* Comment out unused variable declarations

* fix idfilter

* fix doxygen

* reverted ignored setting file

* nop

* merge

* fix warning

---------

Co-authored-by: dvdvgt <david.voigt1998@gmail.com>
Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

SGSSGene committed:
update: adding SQLiteCpp version 3.3.3

SGSSGene committed:
fix: adjust SQLiteCpp CMakeLists.txt

SQLiteCpp is a code copy. To work with OpenMS some minor adjustments
have to be made.
- Bump cmake version, for convenience
- Adjust mechanism to set SQLiteCpp to C++11 or newer
- Set some options to default OFF
    + SQLITECPP_INCLUDE_SCRIPT
    + SQLITECPP_RUN_CPPLINT
    + SQLITECPP_RUN_CPPCHECK
- Remove changing/setting global variables
    + BUILD_SHARED_LIBS
    + CMAKE_RUNTIME_OUTPUT_DIRECTORY
- Delete parts for installing SQLiteCpp into the system

pjones committed:
[DOC] Rewrite build instructions for macOS and Linux

Streamline the build instructions and demote instructions for using
contrib.

timosachsenberg committed:
Fix_pms (#8063)

* some pyopenms fixes

* fix

* still error in pyx

* more fixes

* last fix

* add to ci

* add ci

* Update src/pyOpenMS/pxds/ConsensusMap.pxd

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

---------

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

timosachsenberg committed:
Update cibuild.cmake

pjones committed:
[DOC] Update based on review feedback

pjones committed:
[DOCS/CI] We don't need to pin Python to version 3.13

This is probably an artifact for an older version of macOS.

pjones committed:
[DOC] Stick with Doxygen 1.9 Syntax

We still have to use Doxygen 1.9 on Ubuntu and it doesn't support
indented code fences.

timosachsenberg committed:
Merge pull request #8060 from SGSSGene/fix/update_sqlitecpp

Update SQLiteCpp to 3.3.3

timosachsenberg committed:
Merge branch 'develop' into pjones/ci-install

timosachsenberg committed:
Merge pull request #8042 from OpenMS/pjones/ci-install

[DOC/CI] Make installation documentation and CI use the same commands

timosachsenberg committed:
more fixes

timosachsenberg committed:
Merge pull request #8065 from OpenMS/fix_py

fix: more pyopenms fixes

cbielow committed:
minor runtime improvement (5% for FileConverter) at almost no RAM cost

cbielow committed:
fix CHANGELOG (also for Release 3.4.1)

cbielow committed:
fix compile errors

timosachsenberg committed:
Update cibuild.cmake
## Description

<!-- Please include a summary of the change and which issue is fixed here. -->

## Checklist
- [ ] Make sure that you are listed in the AUTHORS file
- [ ] Add relevant changes and new features to the CHANGELOG file
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] New and existing unit tests pass locally with my changes
- [ ] Updated or added python bindings for changed or new classes (Tick if no updates were necessary.)

### How can I get additional information on failed tests during CI
<details>
  <summary>Click to expand</summary>
If your PR is failing you can check out

- The details of the action statuses at the end of the PR or the "Checks" tab.
- http://cdash.seqan.de/index.php?project=OpenMS and look for your PR. Use the "Show filters" capability on the top right to search for your PR number.
  If you click in the column that lists the failed tests you will get detailed error messages.

</details>

### Advanced commands (admins / reviewer only)
<details>
  <summary>Click to expand</summary>

- `/reformat` (experimental) applies the clang-format style changes as additional commit. Note: your branch must have a different name (e.g., yourrepo:feature/XYZ) than the receiving branch (e.g., OpenMS:develop). Otherwise, reformat fails to push.
- setting the label "NoJenkins" will skip tests for this PR on jenkins (saves resources e.g., on edits that do not affect tests)
- commenting with `rebuild jenkins` will retrigger Jenkins-based CI builds

</details>

---
:warning: Note: Once you opened a PR try to minimize the number of *pushes* to it as every push will trigger CI (automated builds and test) and is rather heavy on our infrastructure (e.g., if several pushes per day are performed).

timosachsenberg committed:
Merge pull request #8071 from OpenMS/luciphor_pep

Use PEP score in LuciphorAdapter

cbielow committed:
use same String hasher across platforms to make result of unordered_map identical

cbielow committed:
revert CV unordered_map. Not portable to MacOS, probably due to differnent bucket sizes or insertion strategy)

cbielow committed:
speed improvement of reading of .gz files by ~7% by increasing buffer to 512kb

cbielow committed:
avoid zlib include in header

cbielow committed:
Merge branch 'develop' into cv_map

cbielow committed:
feature: FileFilter can drop metadataarrays now

cbielow committed:
Merge branch 'develop' of https://github.com/OpenMS/OpenMS into mzml_meta_read

cbielow committed:
feature: faster merging of meta value instances

cbielow committed:
faster reading of mzML when additional binary data is present (e.g. for ion mobility)
 - 20% faster on MSVC
 - 40% faster on GCC

cbielow committed:
faster reading of mzML when additional binary data is present (~20%)

cbielow committed:
adapt exception to match ParseError which occured previously when reading with smaller buffer size

cbielow committed:
Merge branch 'cv_map' of https://github.com/cbielow/OpenMS into cv_map

cbielow committed:
change behaviour of MetaInfo::operator+= (overwrite)

cbielow committed:
Merge pull request #8069 from cbielow/cv_map

faster reading of gzipped files

cbielow committed:
Merge branch 'develop' of https://github.com/OpenMS/OpenMS into mzml_meta_read

cbielow committed:
add tests; augment CHANGELOG

cbielow committed:
Merge branch 'mzml_meta_read' of https://github.com/cbielow/OpenMS into mzml_meta_read

# Conflicts:
#	src/openms/include/OpenMS/METADATA/MetaInfo.h
#	src/openms/source/FORMAT/HANDLERS/MzMLHandler.cpp

cbielow committed:
fix bilinear interpolation test: avoid randomly using an identical matrix

pjones committed:
[DOCKER] Don't use `git submodule` unless necessary

This change allows users to build the Docker container without needing
to have a copy of the git repository.  As long as they have a complete
copy of the third party repo locally everything will just work.

cbielow committed:
build static sqlite libs on Windows

cbielow committed:
Merge branch 'develop' into mzml_meta_read

cbielow committed:
Merge pull request #8079 from cbielow/fix_sql

build static sqlite libs on Windows

timosachsenberg committed:
Merge branch 'develop' into mzml_meta_read

cbielow committed:
Merge pull request #8074 from cbielow/mzml_meta_read

Up to 40% faster mzML reading

Commit:
Fix: correct marker ion deduplication by using 'equal_by_mass' comparator

yumengyann committed:
Update src/openms/source/ANALYSIS/NUXL/NuXLParameterParsing.cpp

Co-authored-by: Timo Sachsenberg <timo.sachsenberg@uni-tuebingen.de>

timosachsenberg committed:
Merge pull request #8080 from yumengyann/feature/NuXL

Fix: use equal_by_mass comparator instead of less_by_mass for marker ion deduplicationFeature/nuxl

timosachsenberg committed:
Merge branch 'develop' into feature/NuXL

timosachsenberg committed:
Apply suggestions from code review

timosachsenberg committed:
Update src/openms_gui/source/VISUAL/LayerDataBase.cpp

timosachsenberg committed:
Update executables.cmake

Commit:
fix cmake

Commit:
merge deveop

Commit:
add constructor

Commit:
fix bug

Commit:
fx counting if openmp disabled

timosachsenberg committed:
fix test

jcharkow committed:
OpenSwath: add parameter to adjust non linear irt retention time extraction window (#8084)

* add parameter to adjust non linear irt retention time extraction window

* set min float

timosachsenberg committed:
Merge remote-tracking branch 'origin' into feature/NuXL

timosachsenberg committed:
try atomic

timosachsenberg committed:
add more atomic

timosachsenberg committed:
fix doxygen (#8088)

timosachsenberg committed:
tie braker

timosachsenberg committed:
changelog

timosachsenberg committed:
relax uniqueness

timosachsenberg committed:
maybe fix test

timosachsenberg committed:
lgamma seems to be faster (and maybe more portable)

timosachsenberg committed:
make more deterministic

timosachsenberg committed:
Merge pull request #7815 from OpenMS/feature/NuXL

feature: Nucleotide-Protein search engine NuXL

timosachsenberg committed:
determinism on windows

timosachsenberg committed:
convert hmtl table to markdown (#8092)

timosachsenberg committed:
Better peptide table in protein_quantifier output (#8094)

* quick save

* add additional level for aggregation and better export in protein quantifier

* inital support for extended protein tables

* fix

* fix number of filenames

* still some bug

* better name for abundance variable

* refactor. extracted a few methods

* proper file+channel level output

* update name

* remove debug code

* compile fix

* make file name in output columns consistent

* better separator

* also sample level quant

* add sample names to output

timosachsenberg committed:
fix small bug and some refactoring (#8096)

fixes bug introduced earlier

timosachsenberg committed:
Update PeptideAndProteinQuant.cpp (#8097)

pjones committed:
[GITHUB]: Add issue templates with forms

Use GitHub issue forms to encourage users to report more information
and to direct questions to more appropriate venues.

Avni2000 committed:
Arrow Integration + Test in CI (#8091)

* Adding support for building apache arrow/parquet integration

* Reverting to original suggested changes based on feedback + docs

* ArrowTest

* Fixing weird git merge

* get Arrow_test setup

* Arrow test (compiles!)

* Set Parquet flag to on, github CI changes

* One liner for parquet flag

* Maybe cmake cares about where the flag is? Idk just a try.

* Json on?

* committing to fetch!

* Commit after all the cursed merges.

* Fixed cmake for nlohmann

* Set implicit conversion to off

* using a set variable?

* force

* revert to normal, purge nlohmann right after install.

* sleeeep

* It's always the one liners

* explicit conversion

* from_json overload

* whoops! double declaration.

* Add arrow to win/mac in CI

* turn external json off only for mac. Unsure of windows.

* doing it right this time - mac cmake flag

* try self closing tags?

* warnings -> errors

* external stuff

* reverting some unnecessary (I THINK) changes

* Unsure why it doesn't read env.WITH_PARQUET

* revert to this if the next idea fails

* checking if this works for the pr

* Works + fixing windows' external json

* learned what git revert does LOL

This reverts commit a93afcf31728009fbff98826af65f78f3abb280b.

* Revert "checking if this works for the pr"

This reverts commit df44c6ac30c38fc58e2e4ad7cf17bbc2c88f063e.

* back to normal

* detailed comment

* small changes as per PR

* Arrow on Conda

poshul committed:
Merge branch 'develop' into cleanup_pyopenms_wheels
